swap/src/cli/list_sellers.rs

```diff
@@ -164,30 +165,42 @@ impl EventLoop {
                                 self.reachable_asb_address.insert(peer_id, address.clone());
                             }
                         }
-                        SwarmEvent::UnreachableAddr { peer_id, error, address, .. } => {
-                            if address == self.rendezvous_addr {
+                        SwarmEvent::OutgoingConnectionError { peer_id, error } => {
+                            if let Some(peer_id_from_error) = peer_id {
+
+                                if &peer_id_from_error == &self.rendezvous_peer_id {
                                 tracing::error!(
                                     "Failed to connect to rendezvous point at {}: {}",
-                                    address,
+                                    &self.rendezvous_addr,
                                     error
                                 );
 
                                 // if the rendezvous node is unreachable we just stop
                                 return Vec::new();
+                            } else {
+                                tracing::error!(
+                                    "You connected to the wrong Peer: {} Error: {}",
```

Comment:
I was so confused because of the indentation :D

Do you know if these events are also called for connections with asbs?
If they are we didn't really fail to connect to a wrong peer but we failed to connect to an asb.

Reply:
Yeah it's awful that rustfmt fails in the select! macro. Updated this indentation correctly here: dc68344b62db029809a4999c11a645238e7c6ad7 

In general, we want to keep as little logic as possible in the select macro and rather call a function with the event. I'll do that update sometime after this goes in so as not to complicate this PR too much.